src/magister.js

```diff
@@ -311,13 +311,20 @@ class Magister {
 	}
 
 	/**
+	 * @param {Integer} [parentId = 0]
 	 * @return {Promise<FileFolder[]>}
 	 */
-	fileFolders() {
+	fileFolders(parentId = 0) {
 		return this._privileges.needs('bronnen', 'read')
-		.then(() => this.http.get(`${this._personUrl}/bronnen?soort=0`))
+		.then(() => {
+			if (parentId === 0) {
+				return this.http.get(`${this._personUrl}/bronnen?soort=0`)
+			} else {
+				return this.http.get(`${this._personUrl}/bronnen?soort=0&parentId=${parentId}`)
+			}
+		})
 		.then(res => res.json())
-		.then(res => res.Items.map(f => new FileFolder(this, f)))
+		.then(res => res.Items.filter(item => item.BronSoort === 0).map(f => new FileFolder(this, f)))
```

Comment:
Does filefolders have `BronSoort`s? I mean I can't check, but just to make sure, sounds a bit weird (but what can I expect from Magister).

Reply:
The filter is applied before the FileFolder object is made, so it looks at the JSON data from the Magister API. As far as I can tell a folder always has `BronSoort` set to `0` in Magister and a `UniqueId` of `00000000-0000-0000-0000-000000000000`. This goes for all folders (gedeeld, prullenbak, subfolders, etc.).

I just noticed that there is also a `BronType` parameter available which gives more detailed information on what we are dealing with. Let me see if I can get a better filter out of that one.

Reply:
Alright, what I ended up doing is checking if something is a file. I used this data as a reference:
```
0: "Onbekend"
1: "Document"
2: "Link"
4: "Scorm"
8: "MijnDocumentenMap"
16: "GedeeldeDocumentenMap"
32: "PortfolioDocumentenMap"
64: "ProjectDocumentenMap"
128: "PrullenbakMap"
256: "Map"
512: "Kennisnet"
2048: "OneDriveForBusiness"
```

In case you're wondering why I didn't check if it is a map, that's simply because the types the API returns didn't match the types that are listen in the list above (in the case of maps, files seem to be fine).